### DISCUSSION TITLE
Can we add custom edges like freestyle edge?

### DISCUSSION THREAD

**Discussion by TahsinTariq** [bnpr/Malt#63] [Feedback & Feature Requests]
I was thinking we could have a feature just like `mark freestyle edge` but for malt. How tough would that be to implement? And does anyone else feel like it would be useful?

**Comment by pragma37** [comment]
This has been requested a few times.

Freestyle and Malt are VERY different, but you can get the same results with a different workflow.
In Malt you can get lines between ID boundaries and you can drive the ID value with whatever you want (textures, vertex colors, uvs, procedural textures...)
In this scene, for example, I use a mask texture for the clothing ornaments and I set the PO.id based on that mask, so it gets lines rendered around them:
https://twitter.com/pragma37/status/1361726754533031936

The closest workflow would be using vertex colors instead of textures.

I guess it should be possible to make an operator that sets a different vertex color on each side of the selected edges, that way you would get basically the same workflow.

**Reply by TahsinTariq** [reply]
Well, this might be a stupid question, but I want to know how the lines are rendered. I mean how are they detected? Does it come from the UV map? Is that why the lines glitch when you scale them but the scale isn't applied?

**Reply by pragma37** [reply]
Here's the line detection code:
https://github.com/bnpr/Malt/blob/master/Malt/Shaders/Filters/Line.glsl
And here's the line compositing code:
https://github.com/bnpr/Malt/blob/master/Malt/Shaders/Passes/LineComposite.glsl#L148

It all happens in screen space.

>  Is that why the lines glitch when you scale them but the scale isn't applied?

Could you report that as an issue? Ideally with a blend file example.

**Reply by TahsinTariq** [reply]
Sorry for the delay.
I created an issue at #70 

And thanks for the links. I'll be looking into them.
